### PR TITLE
doc: use sphinx-gihub-role package

### PR DESCRIPTION
**Pull request by sappelhoff**
I think this would be a neat addition, similar to [gh_substitutions.py](https://github.com/mne-tools/mne-bids/blob/4c0755c6eb567c7840d70acf06f22201c5e75c65/doc/sphinxext/gh_substitutions.py#L1) in mne-bids, but without having to add an extra file.

Changelog was updated using regexps in VSCode search&replace:

- find:
```
`#(\d+) <https://github.com/autoreject/autoreject/pull/\d+>`_
```
- replace:
```
:github:`#$1`
```

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -29,6 +29,9 @@
     'sphinx_copybutton',
```
Comment (astrojuanlu) [inline doc/conf.py:29]: This is missing `sphinx_github_role` in the list of extensions :)

Comment (sappelhoff) [reply doc/conf.py:29]: gosh :man_facepalming:  ... thanks a lot @astrojuanlu

**Comment by astrojuanlu** [comment]
Also, notice you can enable the RTD PR builder to see the docs live! https://blog.readthedocs.com/pull-request-builder-general-availability/

**Comment by sappelhoff** [comment]
> Also, notice you can enable the RTD PR builder to see the docs live! https://blog.readthedocs.com/pull-request-builder-general-availability/

I know, I use this feature in lots of other repos and I love it! :-) I think we don't have it here because we host the docs on GitHub Pages and don't want duplicate locations.

**Comment by jasmainak** [comment]
Thanks a lot @sappelhoff ! This is fantastic :)

**Pull request merged**